float8_experimental/inference.py

```diff
@@ -0,0 +1,257 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD 3-Clause license found in the
+# LICENSE file in the root directory of this source tree.
+"""
+Defines an nn module designed to be used during inference
+"""
+from dataclasses import dataclass
+
+from enum import auto, Enum
+from typing import Callable, List, Optional
+
+import torch
+import torch.nn as nn
+
+from float8_experimental.float8_tensor import (
+    Float8Tensor,
+    ScaledMMConfig,
+    tensor_already_casted_to_fp8,
+    to_fp8_no_autograd,
+)
+from float8_experimental.float8_utils import e4m3_dtype, tensor_to_scale
+
+
+class ActivationCasting(Enum):
+    """Types of quantization to perform on the activations
+
+    WEIGHT_ONLY: Only quantize the weight, no activation casting, weight will be dequantized in the forward pass
+    STATIC: Activation is quantized during model initialization with a static scale
+    DYNAMIC: Activation is quantized during forward pass with a dynamic scale calculated from the input activation
+    """
+
+    WEIGHT_ONLY = auto()
```

Comment:
nit: maybe this is `NONE`, since "weight-only" is not doing anything to activation?

Reply:
I agree what you said makes more sense, but I do think that "weight only" or "wo" has become the more common term

Reply:
oh, I was just pointing out that `weight_only` is about the weight, but the current class name is `ActivationCasting`

Reply:
Totally agree, I think we wanna unify these names with torchao anyways, is it okay if I leave it as is for this PR?

Reply:
sgtm, maybe a TODO to fix the name later?